bookkeeper-dist/src/main/resources/LICENSE-all.bin.txt

```diff
@@ -202,507 +202,265 @@
+[26] Source available at https://github.com/eclipse/vert.x/tree/3.4.1
+[27] Source available at https://github.com/vert-x3/vertx-web/tree/3.4.1
+[28] Source available at https://github.com/javax-inject/javax-inject/tree/1
+[29] Source available at http://logging.apache.org/log4j/1.2/download.html
+[30] Source available at https://github.com/java-native-access/jna/tree/3.2.7
+[31] Source available at https://git-wip-us.apache.org/repos/asf?p=commons-collections.git;a=tag;h=a3a5ad
+[32] Source available at https://git-wip-us.apache.org/repos/asf?p=commons-lang.git;a=tag;h=3ad2e8
+[33] Source available at https://github.com/apache/zookeeper/tree/release-3.5.3
+[34] Source available at https://github.com/eclipse/jetty.project/tree/jetty-9.4.5.v20170502
 
 ------------------------------------------------------------------------------------
-For lib/com.google.protobuf-protobuf-java-3.4.0.jar
-
-Copyright 2014, Google Inc.  All rights reserved.
-
-Redistribution and use in source and binary forms, with or without
-modification, are permitted provided that the following conditions are
-met:
-
-    * Redistributions of source code must retain the above copyright
-      notice, this list of conditions and the following disclaimer.
-    * Redistributions in binary form must reproduce the above
-      copyright notice, this list of conditions and the following disclaimer
-      in the documentation and/or other materials provided with the
-      distribution.
-    * Neither the name of Google Inc. nor the names of its
-      contributors may be used to endorse or promote products derived from
-      this software without specific prior written permission.
-
-THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
-Code generated by the Protocol Buffer compiler is owned by the owner
-of the input file used when generating it.  This code is not
-standalone and requires a support library to be linked with it.  This
-support library is itself covered by the above license.
+lib/io.netty-netty-3.10.1.Final.jar contains the extensions to Java Collections Framework which has
+been derived from the works by JSR-166 EG, Doug Lea, and Jason T. Greene:
+
+  * LICENSE:
+    * deps/netty-3.10.1.Final/LICENSE.jsr166y.txt (Public Domain)
```

Comment:
I am not sure we need to retain these LICENSE files. I would remove them and just reference back to netty. 

Also I think netty is used by multiple ASF projects, it would be good if you can point me any ASF projects that do this. Otherwise it is going to be painful to maintain LICENSE of those "compiled code" in our dependencies.

Reply:
If we include the code, we should include the licenses. The public domain ones are not so important, but the BSD style ones need to be either here, or in a license file linked from here. Dependencies of dependencies should be treated the same as dependencies [1], which makes sense because if netty is distributing it, then we are distributing it.

[1] http://www.apache.org/dev/licensing-howto.html#deps-of-deps

Reply:
if a dependency of dependency is bundled, it makes sense to include its license. otherwise it doesn't make sense to me. 

I am not sure about "code copied" into a netty and we bundle netty as a dependency. It is a very complicated case to me. I would suggest following what other ASF projects that use Netty do. I am pretty sure there are tons of projects depend on netty. 

If you really feel strong about using this approach you proposed here, I would suggest reaching legal-discuss@ to get a clarification on how to update LICENSE and NOTICE when including a bundle that contains copied-and-modified code. I am not qualified for reviewing if this is the right or not.

From my personal view, I would follow what other ASF projects do, which they are more or less reviewed and accepted by the ASF. The approach here suddenly pulls in a lot of licenses from dependencies of dependencies, sounds too complicated to me and it is going to be very hard for the community to maintain and keep them in sync.

Reply:
Code copied is actually the simplest case. You are distributing binaries derived from from copied code, and if it is BSD/MIT licensed, then you must include the license. Since I have the hadoop NOTICE open, I'll use that as the example:
https://github.com/apache/hadoop/blob/trunk/NOTICE.txt
Netty is there at the top (though the BSD/MIT stuff should be in LICENSE I believe).

I've emailed legal-discuss@, you're cc'd.

Yes, this is a pain, but it's less of a pain than reimplementing all out dependencies ourselves. And it's not that complex. I'll write it up if this goes in, and I don't expect it to be more than 2 or 3 paragraphs.

> sounds too complicated to me and it is going to be very hard for the community to maintain and keep them in sync.

The other option is stalled release processes if we leave it up to then to check licenses. That's what I'm trying to get away from with this.

Reply:
since you point out the hadoop's NOTICE file as the example, can you please tell me where does hadoop keep the those license files of netty's dependencies? (e.g. license/LICENSE.jsr166y.txt, license/LICENSE.base64.txt) I can't find those license files in hadoop repo or their binary package.

If I understand this correctly, hadoop only adds netty's notice file in its notice file, it doesn't copy the licenses of netty's dependencies into its repo and redistributed with its binary package. https://github.com/netty/netty/blob/4.1/NOTICE.txt The approach is exactly what I said here, we don't need to copy the licenses of netty's dependencies, we only need to attach the notices/licenses of the dependencies we bundled.

If you think hadoop is the right example to follow, let's follow what hadoop does (remove all the deps licenses, only include the notice/license of the bundles we distribute). If you think hadoop's notice/license also has problems, let's wait for responses from the legal team.

Reply:
I do think hadoop has problems in this regard. The jsr166y and base64 being missing aren't a problem, since they're public domain, but the others BSD stuff should have the license distributed.

Reply:
Let's wait to see what legal say

Reply:
sure. let's see

Reply:
Any updates from legal?

Reply:
pinged the list again.

Reply:
http://mail-archives.apache.org/mod_mbox/www-legal-discuss/201801.mbox/ajax/%3CCAJdLeK0euKouMx2h2%2BK6EC2-HJSL7XZnHcA7qvDtejN-D4k0QA%40mail.gmail.com%3E

Reply:
@sijie legal's position on this is "if it's bundled, it needs the license"

Reply:
@ivankelly I left one comment in the email thread, will wait for a clarification of my understanding there.